src/pocketmine/block/EnderChest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\item\Item;
+use pocketmine\item\ItemFactory;
+use pocketmine\item\Tool;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Vector3;
+use pocketmine\Player;
+use pocketmine\tile\EnderChest as TileEnderChest;
+use pocketmine\tile\Tile;
+
+class EnderChest extends Transparent{
+
+	protected $id = self::ENDER_CHEST;
+
+	public function __construct(int $meta = 0){
+		$this->meta = $meta;
+	}
+
+	public function getHardness() : float{
+		return 22.5;
+	}
+
+	public function getBlastResistance() : float{
+		return 3000;
+	}
+
+	public function getName() : string{
+		return "Ender Chest";
+	}
+
+	public function getToolType() : int{
+		return Tool::TYPE_PICKAXE;
+	}
+
+	protected function recalculateBoundingBox() : ?AxisAlignedBB{
+		return new AxisAlignedBB(
+			$this->x + 0.0625,
+			$this->y,
+			$this->z + 0.0625,
+			$this->x + 0.9375,
+			$this->y + 0.9475,
```

Comment:
is this a typo or intended?

Reply:
typo, will fix.

Reply:
according to tests with MCPE, the bounding boxes are actually +0.05 and +0.95 offset, with +0.95 height, not 0.0625. You get kicked for flying when walking on top of chests. That will need to be fixed separately.

Regardless, this class should be able to extend Chest, no?

Reply:
actually that doesn't even make sense... I see 0.0625 in IDA, but my actual in-game collision shows 0.05. Needs more investigation.

Reply:
Making the class extend chest should fix it, one would think it safe to assume chest and ender chests would share the same geometric properties.

Reply:
yeah, their bbs look the same
![cd](https://user-images.githubusercontent.com/15074389/31572410-29018292-b0ae-11e7-9c83-3255e7990726.PNG)